controllers/front/OrderConfirmationController.php

```diff
@@ -45,31 +48,92 @@ public function init()
     {
         parent::init();
 
+        // If are coming to this page to finish free order, we need to run additional logic
         if (true === (bool) Tools::getValue('free_order')) {
             $this->checkFreeOrder();
         }
 
-        $this->id_cart = (int) (Tools::getValue('id_cart', 0));
-
-        $redirectLink = 'index.php?controller=history';
-
+        // We get the required data from URL
```

Comment:
> I am pretty rich on coments, if it's not a good practice, I will reduce it, but this way it can be read like a book.

Actually, it should be a code that you read like a book! :) For example in this place, it's obvious that we're getting data from an URL, same for checking password length, etc.

Reply:
Whether comments are good or not has been a discussion for dozen of years among developers.

Famous quote by Robert C Martin (author of Clean Code):
> A comment is a failure to express yourself in code. If you fail, then write a comment; but try not to fail.

But plenty of people, including others renowned authors have disagreed with him.

I don't have the Ultimate Truth but I think @kpodemski opinion is a good one: comments should bring some informations to the reader that he could not find easily from the code. If you can grasp an information either from the code (getting data from an URL) or from a comment, the comment is not useful. Look at @jolelievre comments in PR https://github.com/PrestaShop/PrestaShop/pull/24991/ they give such information.

However, when you have to choose whether to add a comment or not to add, I think it's always good to add it :) it's better to have a code that is uselessly too obvious rather than an obscure code 😄 

That being said... I'm quite sure some of the other maintainers will disagree with me :trollface:

Reply:
I agree that it's preferable to risk having redundant comments rather than lose valuable contextual information. Don't make me reverse engineer code!